Tools/auterion/remote_update_fmu.sh

```diff
@@ -50,53 +50,66 @@ while getopts ":f:c:d:p:u:" opt; do
 				exit 1
 			fi
 			;;
+		r )
+			revert=true
+			;;
 	esac
 done
 
-target_dir=/shared_container_dir/fmu
+if [ -z "$device" ]; then
+	echo "Error: missing device"
+	exit 1
+fi
 
-# create update-dev.tar
-target_file_name="update-dev.tar"
-tmp_dir="$(mktemp -d)"
-config_path=""
-firmware_path=""
+if [ "$revert" == true ]; then
+	# revert to the release version which was originally deployed
+	cmd="cp /persistent/shared_container_dir/fmu/update.tar /persistent/shared_container_dir/fmu/update-dev.tar"
+	ssh -t -p $ssh_port $ssh_user@$device "$cmd"
+else
+	# create custom update-dev.tar
+	tmp_dir="$(mktemp -d)"
+	config_path=""
+	firmware_path=""
 
-if [ -d "$config_dir" ]; then
-	cp -r "$config_dir" "$tmp_dir/config"
-	config_path=config
-fi
+	if [ -d "$config_dir" ]; then
+		cp -r "$config_dir" "$tmp_dir/config"
+		config_path=config
+	fi
 
-if [ -f "$firmware_file" ]; then
-	extension="${firmware_file##*.}"
-	cp "$firmware_file" "$tmp_dir/firmware.$extension"
-	if [ "$extension" == "elf" ]; then
-		# ensure the file is stripped to reduce file size
-		arm-none-eabi-strip "$tmp_dir/firmware.$extension"
+	if [ -f "$firmware_file" ]; then
+		extension="${firmware_file##*.}"
+		cp "$firmware_file" "$tmp_dir/firmware.$extension"
+		if [ "$extension" == "elf" ]; then
+			# ensure the file is stripped to reduce file size
+			arm-none-eabi-strip "$tmp_dir/firmware.$extension"
+		fi
+		firmware_path="firmware.$extension"
 	fi
-	firmware_path="firmware.$extension"
-fi
 
-if [ -z "$device" ]; then
-	echo "Error: missing device"
-	exit 1
-fi
+	pushd "$tmp_dir" &>/dev/null
 
-pushd "$tmp_dir" &>/dev/null
+	if [ -z $firmware_path ] && [ -z $config_path ]; then
+		exit 1
+	fi
 
-if [ -z $firmware_path ] && [ -z $config_path ]; then
-	exit 1
-fi
+	tar_name="tar"
+
+	if [ -x "$(command -v gtar)" ]; then
+		# check if gnu-tar is installed on macOS and use that instead
+		tar_name="gtar"
+	fi
+
+	target_dir=/shared_container_dir/fmu
+	target_file_name="update-dev.tar"
+
+	$tar_name -C "$tmp_dir" --sort=name --owner=root:0 --group=root:0 --mtime='2019-01-01 00:00:00' -cvf $target_file_name $firmware_path $config_path
 
-# check if gnu-tar is installed on macOS and use that instead
-tar_name="tar"
-if [ -x "$(command -v gtar)" ]; then
-	tar_name="gtar"
+	# send it to the target to start flashing
+	scp -P $ssh_port "$target_file_name" $ssh_user@"$device":$target_dir
+	popd &>/dev/null
+	rm -rf "$tmp_dir"
 fi
-$tar_name -C "$tmp_dir" --sort=name --owner=root:0 --group=root:0 --mtime='2019-01-01 00:00:00' -cvf $target_file_name $firmware_path $config_path
-scp -P $ssh_port "$target_file_name" $ssh_user@"$device":$target_dir
-popd &>/dev/null
-rm -rf "$tmp_dir"
 
-# grab status output
+# grab status output for flshing progress
```

Comment:
flashing

Reply:
Thanks for spotting 🤦‍♂️ Corrected in https://github.com/PX4/PX4-Autopilot/commit/cdc784eeb3b5257fcb6f38406705224748b88dbd